guvnor-ala/guvnor-ala-services-api/src/main/java/org/guvnor/ala/services/api/RuntimeProvisioningService.java

```diff
@@ -0,0 +1,145 @@
+
+/*
+ * Runtime Provisioning Service interface. It allows us to register and interact with different Provisioning Providers and
+ *  to create new Runtimes on these providers.
+ * URL: {app-context}/runtime/
+ */
+@Path( "runtime" )
+public interface RuntimeProvisioningService {
+
+    /*
+     * Get all the registered ProviderTypes
+     * @return ProviderTypeList containing all the registered provider types
+     * @see ProviderTypeList
+     * @throw BusinessException in case of an internal exception
+     */
+    @GET
+    @Consumes( value = APPLICATION_JSON )
+    @Produces( value = APPLICATION_JSON )
+    @Path( "providertypes" )
+    ProviderTypeList getAllProviderTypes() throws BusinessException;
+
+    /*
+     * Get all the registered Providers
+     * @return ProviderList containing all the registered providers
+     * @see ProviderList
+     * @throw BusinessException in case of an internal exception
+     */
+    @GET
+    @Produces( value = APPLICATION_JSON )
+    @Path( "providers" )
+    ProviderList getAllProviders() throws BusinessException;
+
+    /*
+     * Register a new Provider
+     * @param ProviderConfig used to create the new Provider
+     * @throw BusinessException in case of an internal exception
+     */
+    @POST
+    @Consumes( value = APPLICATION_JSON )
+    @Path( "providers" )
+    void registerProvider( @NotNull ProviderConfig conf ) throws BusinessException;
+
+    /*
+     * Unregister an existing Provider
+     * @param String provider name
+     * @throw BusinessException in case of an internal exception
+     */
+    @DELETE
+    @Path( "providers" )
+    void unregisterProvider( @FormParam( value = "name" ) String name ) throws BusinessException;
+
+    /*
+     * Create a new Runtime
+     * @param RuntimeConfig containing the configuration used to create the new Runtime
+     * @throw BusinessException in case of an internal exception
+     */
+    @POST
+    @Path( "runtimes/" )
+    @Consumes( value = APPLICATION_JSON )
+    String newRuntime( @NotNull RuntimeConfig conf ) throws BusinessException;
+
+    /*
+     * Destroy an existing  Runtime
+     * @param String runtimeId of the runtime to destroy
+     * @throw BusinessException in case of an internal exception
+     */
+    @DELETE
+    @Path( "runtimes/{id}" )
+    void destroyRuntime( @PathParam( value = "id" ) String runtimeId ) throws BusinessException;
+
+    /*
+     * Get All Runtimes
+     * @return RuntimeList containing all the registered Runtimes 
+     * @throw BusinessException in case of an internal exception
+     */
+    @GET
+    @Produces( value = APPLICATION_JSON )
+    @Path( "runtimes/" )
+    RuntimeList getAllRuntimes() throws BusinessException;
+
+    /*
+     * Start a given Runtime
+     * @param String runtimeId of the runtime to be started
+     * @throw BusinessException in case of an internal exception
+     */
+    @POST
+    @Path( "runtimes/{id}/start" )
```

Comment:
similar comment here, could we replace the start/stop/restart with other non verb endpoints and maybe with help of HTTP methods - POST, PUT, DELETE?